src/iamserver/nodecontroller.cpp

```diff
@@ -355,10 +323,33 @@ void NodeController::Close()
 grpc::Status NodeController::HandleRegisterNodeStream(const std::vector<aos::NodeStatus>& allowedStatuses,
     NodeServerReaderWriter* stream, grpc::ServerContext* context, aos::iam::nodemanager::NodeManagerItf* nodeManager)
 {
-    auto handler = std::make_shared<NodeStreamHandler>(allowedStatuses, stream, context, nodeManager);
-    StoreHandler(handler);
+    auto handler = std::make_shared<NodeStreamHandler>(stream, context);
+
+    auto onNodeInfoReceivedCallback
```

Comment:
Do we really need to pass it as callback? Can't it be private method?

Reply:
Refactored